pkg/docker/docker_test.go

```diff
@@ -24,201 +30,336 @@ func TestContainerName(t *testing.T) {
-		},
-		expectedImage: nil,
-		expectedError: errors.InspectImageError,
-	}
-	pullErrorTest := testDef{
-		imageName: "test_image",
-		docker: test.FakeDockerClient{
-			PullImageErr:       docker.ErrConnectionRefused,
-			InspectImageErr:    []error{nil},
-			InspectImageResult: []*docker.Image{nil},
-		},
-		expectedImage:       nil,
-		expectedError:       errors.PullImageError,
-		expectedPullOptions: docker.PullImageOptions{Repository: "test_image:" + DefaultTag},
-	}
-	errorAfterPullTest := testDef{
-		imageName: "test_image:testtag",
-		docker: test.FakeDockerClient{
-			InspectImageErr:    []error{docker.ErrNoSuchImage, docker.ErrNoSuchImage},
-			InspectImageResult: []*docker.Image{nil, image},
-		},
-		expectedImage:       nil,
-		expectedError:       errors.InspectImageError,
-		expectedPullOptions: docker.PullImageOptions{Repository: "test_image:testtag"},
+		imageName:     "test_image",
+		docker:        test.FakeDockerClient{},
+		calls:         []string{"inspect_image", "pull"},
+		expectedError: "unable to get test_image:latest",
 	}
 	tests := map[string]testDef{
-		"ImageExists":    imageExistsTest,
-		"ImagePulled":    imagePulledTest,
-		"InspectError":   inspectErrorTest,
-		"PullError":      pullErrorTest,
-		"ErrorAfterPull": errorAfterPullTest,
+		"ImageExists":  imageExistsTest,
+		"InspectError": inspectErrorTest,
 	}
 
 	for test, def := range tests {
 		dh := getDocker(&def.docker)
+		fake := dh.kubeDockerClient.(*dockertools.FakeDockerClient)
+		if len(def.expectedError) > 0 {
+			fake.ClearErrors()
+			fake.InjectError("pull", fmt.Errorf(def.expectedError))
+			fake.Image = nil
+		} else {
+			fake.Image = &dockertypes.ImageInspect{ID: def.imageName}
+		}
+
 		resultImage, resultErr := dh.CheckAndPullImage(def.imageName)
-		if resultImage != def.expectedImage {
-			t.Errorf("%s: Unexpected image result -- %v", test, resultImage)
+
+		if e := fake.AssertCalls(def.calls); e != nil {
+			t.Errorf("%s %+v", test, e)
 		}
-		if e, ok := resultErr.(errors.Error); def.expectedError != 0 && (!ok || e.ErrorCode != def.expectedError) {
+
+		if len(def.expectedError) > 0 && (resultErr == nil || resultErr.Error() != def.expectedError) {
 			t.Errorf("%s: Unexpected error result -- %v", test, resultErr)
 		}
-		pullOpts := def.docker.PullImageOpts
-		if !reflect.DeepEqual(pullOpts, def.expectedPullOptions) {
-			t.Errorf("%s: Unexpected pull image opts -- %v", test, pullOpts)
+		if fake.Image != nil && (resultImage == nil || resultImage.ID != def.imageName) {
+			t.Errorf("%s: Unexpected image result -- %+v instead of %+v", test, resultImage, fake.Image)
 		}
 	}
 }
 
 func TestRemoveContainer(t *testing.T) {
 	fakeDocker := &test.FakeDockerClient{}
 	dh := getDocker(fakeDocker)
+	fake := dh.kubeDockerClient.(*dockertools.FakeDockerClient)
 	containerID := "testContainerId"
-	expectedOpts := docker.RemoveContainerOptions{
-		ID:            containerID,
-		RemoveVolumes: true,
-		Force:         true,
```

Comment:
so the kube-client's RemoveContainer isn't setting/passing these down to the docker client?